### ISSUE TITLE
Slow MapEntity updates

### ISSUE THREAD

**Issue by rekterakathom** [Regalis11/Barotrauma#11642]
## Problem
Map Entity updates take a long time, and cause performance regression. [A community mod](mapEntityUpdateInterval) has been made to combat this by increasing the time between map entity updates. I feel like this could be improved in the game itself. The mod is doubling performance for some users, so this might be a high priority performance update.

## Proposed solution
Add a system that updates entities based on their distance from players, and maybe item priority. The workshop mod is getting away with reducing the update times (almost) across the board by a factor of 4, so I feel like that's a good scale. Make 4 distance ranges and adjust entity update rate accordingly. Some items and entities need to be always updated though, so exclude these.

I have not delved very deep into the codebase to see if anything like this is already being done. I am willing to start investigating and working on this issue, but I have no experience with the Barotrauma codebase nor C#. If someone is willing to co-author a PR for this with me and tutor me a bit, feel free to contact me.

**Comment by Regalis11** [comment]
Thank you for the report!

We are familiar with the performance fix mod, but unfortunately we feel it's not a robust enough way to address the issue. Updating items only every few frames causes various side-effects (some of which the creator of the mod has addressed by forcing specific items to update every frame). It's non-trivial to determine which items need to update every frame and which don't, especially when it comes to modded items - we cannot expect a modder to be familiar enough with the inner workings of the codebase to determine what kind of an update rate is required for their item.

[We've got some optimizations coming up](https://github.com/Regalis11/Barotrauma-development/tree/fix/material-hoarding-optimizations) though that will provide similar performance benefits though, although we took a slightly different approach: we instead automatically detect certain situations in which specific items don't need to be updated at all, and put them "to sleep".
